islandora-connector-derivative/src/main/java/ca/islandora/alpaca/connector/derivative/DerivativeConnector.java

```diff
@@ -100,26 +100,66 @@ public void configure() {
 
             .log(DEBUG, LOGGER, "Received message on IslandoraConnectorDerivative-" + connectorName)
 
-            // Parse the event into a POJO.
-            .unmarshal().json(JsonLibrary.Jackson, AS2Event.class)
-
-            // Stash the event on the exchange.
+            // Stash the original JSON body on the exchange.
             .setProperty("event").simple("${body}")
 
+            // Parse the event into a POJO for property extraction.
+            .unmarshal().json(JsonLibrary.Jackson, AS2Event.class)
+            .setProperty("eventPojo").simple("${body}")
+
             // Make the Crayfish request.
             .removeHeaders("*", "Authorization")
             .setHeader(Exchange.HTTP_METHOD, constant("GET"))
-            .setHeader("Accept", simple("${exchangeProperty.event.attachment.content.mimetype}"))
-            .setHeader("X-Islandora-Args", simple("${exchangeProperty.event.attachment.content.args}"))
-            .setHeader("Apix-Ldp-Resource", simple("${exchangeProperty.event.attachment.content.sourceUri}"))
+            .process(exchange -> {
+            final String jsonEvent = exchange.getProperty("event", String.class);
+                if (jsonEvent != null) {
+                    final String b64 = java.util.Base64.getEncoder()
+                    .encodeToString(jsonEvent.getBytes(java.nio.charset.StandardCharsets.UTF_8));
+                    exchange.getIn().setHeader("X-Islandora-Event", b64);
+                }
+            })
+            .process(exchange -> {
+            final AS2Event event = exchange.getProperty("eventPojo", AS2Event.class);
```

Comment:
Indentation seems off?

```suggestion
                final AS2Event event = exchange.getProperty("eventPojo", AS2Event.class);
```